src/librepgp/stream-armor.c

```diff
@@ -908,40 +904,37 @@ init_armored_dst(pgp_dest_t *dst, pgp_dest_t *writedst, pgp_armored_msg_t msgtyp
 bool
 is_armored_source(pgp_source_t *src)
 {
-    const char armor_start[] = "-----BEGIN PGP";
-    uint8_t    buf[128];
-    ssize_t    read;
+    uint8_t buf[128];
+    ssize_t read;
 
     read = src_peek(src, buf, sizeof(buf));
-    if (read < (ssize_t) sizeof(armor_start)) {
+    if (read < (ssize_t) strlen(ST_ARMOR_BEGIN) + 1) {
         return false;
     }
 
     buf[read - 1] = 0;
-    return !!strstr((char *) buf, armor_start);
+    return !!strstr((char *) buf, ST_ARMOR_BEGIN);
```

Comment:
We don't need the double logical not (`!!`) when returning a C99 bool, but it's not a big deal.